### PR TITLE
Allow resolving `enso://` URIs in `Data.read` and other places

### LINKED ISSUE #9048: Resolve `enso://` URIs in `Data.read`
We want to be able to resolve read an Enso file by an URI, e.g. `Data.read "enso://radeusgd/my_dir/foo.txt"`.

To do so, we want to:
- [x] implement resolution of an `enso://` path to an `Enso_File` instance,
  - Currently it only has to work for existing files. For a non-existing file it will return `File_Error.Not_Found`.
- [x] integrate `Enso_File` path resolution with the `FileSystemSPI` and register it for the `enso://` protocol,
- [x] add tests for `Data.read` using it.

### PR DESCRIPTION
**Pull request by radeusgd**
### Pull Request Description

- Implements the core parts of #9048
- Currently the path resolution is done by resolving each segment, one by one - requiring as many API calls as there are segments in the path.
	- This should be replaced in a followup PR, once https://github.com/enso-org/cloud-v2/issues/899 is implemented.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Enso_Secret.enso [hunk 1]
```diff
@@ -70,12 +71,25 @@ type Enso_Secret
...
+       - name: The name of the secret, or an `enso://` path.
+       - parent: The parent folder for the secret, if resolving by name.
+         If `Nothing` then will search in the current working directory.
+         If an `enso://` path is provided, the parent argument is not considered
+         and must be `Nothing`.
     get : Text -> Enso_File | Nothing -> Enso_Secret ! Not_Found
     get name:Text parent:(Enso_File | Nothing)=Nothing =
-        Enso_Secret.list parent . find s-> s.name == name
+        is_path = name.starts_with "enso://"
+        case is_path of
```
Comment (jdunkerley) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Enso_Secret.enso:82]: don't see point of not putting into case statement.

Comment (jdunkerley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Enso_Secret.enso:82]: given how short the `False` branch is would probably have done `if/then/else` with the not condition but that is just stylistic.

**Review by GregoryTravis — APPROVED** [review]

**Comment by radeusgd** [comment]
Sorry for pinging unrelated reviewers - GitHub did this automatically when changing PR's base branches.

File: std-bits/base/src/main/java/org/enso/base/enso_cloud/EnsoPathFileSystemSPI.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  }
+
+  @Override
+  protected String getTypeName() {
+    return "Enso_File";
+  }
+
+  @Override
+  protected String getProtocol() {
+    return "enso";
```
Comment (JaroslavTulach) [inline std-bits/base/src/main/java/org/enso/base/enso_cloud/EnsoPathFileSystemSPI.java:24]: Where is the specification of this protocol? Shouldn't the behavior be defined somewhere in this PR?

**Review by JaroslavTulach — APPROVED** [review]

File: test/AWS_Tests/src/S3_Spec.enso [hunk 3]
```diff
@@ -549,15 +549,16 @@ add_specs suite_builder =
             # It reads the datalink description and then reads the actual S3 file contents:
             (enso_project.data / "simple.datalink") . read . should_equal "Hello WORLD!"
 
-        #group_builder.specify "should be able to read a data link with custom credentials and secrets" pending=cloud_setup.pending <| cloud_setup.with_prepared_environment <|
-        group_builder.specify "should be able to read a data link with custom credentials and secrets" pending="TODO: reading secrets from path" <| cloud_setup.with_prepared_environment <|
+        group_builder.specify "should be able to read a data link with custom credentials and secrets" pending=cloud_setup.pending <| cloud_setup.with_prepared_environment <|
+            f = replace_username_in_datalink (enso_project.data / "credentials-with-secrets.datalink")
```
Comment (jdunkerley) [inline test/AWS_Tests/src/S3_Spec.enso]: minor nit - but can we not use `f`

**Review by jdunkerley — APPROVED** [review]

File: distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Enso_Secret.enso (hunk 1 above)
Comment (radeusgd) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Enso_Secret.enso:82]: > don't see point of not putting into case statement.

the point was to give this expression a name - for documentation purposes

File: std-bits/base/src/main/java/org/enso/base/enso_cloud/EnsoPathFileSystemSPI.java (hunk 2 above)
Comment (radeusgd) [reply std-bits/base/src/main/java/org/enso/base/enso_cloud/EnsoPathFileSystemSPI.java:24]: We don't really have a proper specification but it's just `enso://<org_name>/<path>`.

I added a short summary in f4944e1a176686b688097141ec276a8275aa46f9

**Pull request merged**